rasa_core/policies/embedding_policy.py

```diff
@@ -0,0 +1,2043 @@
+
+            probs = tf.reverse(conv_probs, axis=[1])
+
+        # Sharpening
+        g_sh = self.gamma_sharp(cell_output)
+
+        powed_probs = tf.pow(probs, g_sh)
+        probs = powed_probs / (
+                tf.reduce_sum(powed_probs, 1, keepdims=True) + 1e-32)
+
+        # set probs for no intents and action_listens to zero
+        if ignore_mask is not None:
+            probs = tf.concat([tf.where(ignore_mask,
+                                        tf.zeros_like(probs[:, :-1]),
+                                        probs[:, :-1]),
+                               probs[:, -1:]], 1)
+        return probs, next_scores_state
+
+
+def _compute_time_attention(attention_mechanism, cell_output, attention_state,
+                            # time is added to calculate time attention
+                            time, timed_ntm, ignore_mask, attention_layer):
+    """Computes the attention and alignments limited by time
+        for a given attention_mechanism.
+
+        Modified helper form tensorflow."""
+
+    scores, _ = attention_mechanism(cell_output, state=attention_state)
+
+    # take only scores form current and past times
+    timed_scores = scores[:, :time + 1]
+    timed_scores_state = attention_state[:, :time]
+    if ignore_mask is not None:
+        timed_ignore_mask = ignore_mask[:, :time]
+    else:
+        timed_ignore_mask = None
+
+    # pass these scores to NTM
+    probs, next_scores_state = timed_ntm(cell_output, timed_scores,
+                                         timed_scores_state,
+                                         timed_ignore_mask)
+
+    # concatenate probs with zeros to get new alignments
+    zeros = tf.zeros_like(scores)
+    # remove current time from attention
+    alignments = tf.concat([probs[:, :-1], zeros[:, time:]], 1)
+
+    # Reshape from [batch_size, memory_time] to [batch_size, 1, memory_time]
+    expanded_alignments = tf.expand_dims(alignments, 1)
+
+    # Context is the inner product of alignments and values along the
+    # memory time dimension.
+    # alignments shape is
+    #   [batch_size, 1, memory_time]
+    # attention_mechanism.values shape is
+    #   [batch_size, memory_time, memory_size]
+    # the batched matmul is over memory_time, so the output shape is
+    #   [batch_size, 1, memory_size].
+    # we then squeeze out the singleton dim.
+    context = tf.matmul(expanded_alignments, attention_mechanism.values)
+    context = tf.squeeze(context, [1])
+
+    if attention_layer is not None:
+        attention = attention_layer(tf.concat([cell_output, context], 1))
+    else:
+        attention = context
+
+    # return current time to attention
+    alignments = tf.concat([probs, zeros[:, time + 1:]], 1)
+    next_attention_state = tf.concat([next_scores_state,
+                                      zeros[:, time + 1:]], 1)
+    return attention, alignments, next_attention_state
+
+
+class TimeAttentionWrapperState(
```

Comment:
because I'm not sure how to do that the original is https://github.com/tensorflow/tensorflow/blob/25c197e02393bd44f50079945409009dd4d434f8/tensorflow/contrib/seq2seq/python/ops/attention_wrapper.py#L969